bin/snyk_tests.sh

```diff
@@ -22,13 +24,56 @@ source bin/common/check_returncode.sh
 # authenticate snyk CLI locally
 run_and_check_returncode "snyk auth $SNYK_AUTHENTICATION_TOKEN"
 
-# Iac run limited to 20 runs a month. TBD Discuss with Jon Velapondi.
-# echo "Scanning Infrastructure as code with Snyk for potential vulnerabilities..."
-# snyk iac test || true
-
-# scan orca website
+# Scan orca website
 echo "Scanning open-source code with Snyk for potential vulnerabilities..."
 cd website
 snyk test -d --severity-threshold=high || true
+cd -
+
+function run_snyk_tests() {
+  echo
+  echo "Running snyk test in $1"
+  echo
+
+  cd $1
+  # remove shared library from the requirement files since snyk cannot scan through it
+  grep -v '../../shared_libraries' requirements-dev.txt >> test-requirements-dev.txt
+  grep -v '../../shared_libraries' requirements.txt >> test-requirements.txt
+  snyk test --severity-threshold=high --file=test-requirements-dev.txt --package-manager=pip || true
+  snyk test --severity-threshold=high --file=test-requirements.txt --package-manager=pip || true
+  # remove test files
+  rm test-requirements*.txt
+}
+export -f run_snyk_tests
+
+function run_snyk_tests_others() {
+  echo
+  echo "Running snyk test in $1"
+  echo
+  cd $1
+  # remove shared library from the requirement files since snyk cannot scan through it
+  grep -v '../shared_libraries' requirements-dev.txt >> test-requirements-dev.txt
+  grep -v '../shared_libraries' requirements.txt >> test-requirements.txt
+  snyk test --severity-threshold=high --file=test-requirements-dev.txt --package-manager=pip || true
+  snyk test --severity-threshold=high --file=test-requirements.txt --package-manager=pip || true
+  # remove test files
+  rm test-requirements*.txt
+}
+export -f run_snyk_tests_others
+
+task_dir_shared_lib=$(ls -d shared_libraries)
+task_dir_graphql=$(ls -d graphql)
+lambda_task_dirs=$(ls -d tasks/*)
+
+# Run snyk for shared library dir
+echo "Running snyk for shared library dir"
+parallel --jobs 0 -n 1 -X --halt now,fail=1 run_snyk_tests_others ::: $task_dir_shared_lib
+
+# Run snyk for graphql dir
+echo "Running snyk for graphql"
+parallel --jobs 0 -n 1 -X --halt now,fail=1 run_snyk_tests_others ::: $task_dir_graphql
```

Comment:
The parallel here doesn't do anything since there is only one directory. I'd personally recommend adding graphql and shared to a single command so you have the parallel command for lambdas, and the parallel command for structures that require run_snyk_tests_others.

Reply:
I am not sure how to do that using parallel..
Do you know the format to scan through multiple directories using parallel?

Reply:
You're already passing in a list, so parallel is already set up to scan multiple directories (See $lambda_task_dirs). You just need to modify the argument at the end of the `parallel` command to have multiple strings.

I ran `lambda_task_dirs=$(ls -d tasks/*)` `echo $lambda_task_dirs` and the format is `tasks/copy_from_archive tasks/copy_to_archive tasks/db_deploy tasks/delete_old_reconcile_jobs` so I'd recommend you create `other_dirs="shared_libraries graphql"` and run parallel as `parallel --jobs 0 -n 1 -X --halt now,fail=1 run_snyk_tests_others ::: $other_dirs`
This also removes the `ls` commands around graphql and shared_libraries, since they only returned "graphql" and "shared_libraries" respectively.

Reply:
cool thanks. It works!